layout/_widget/page-timeline.ejs

```diff
@@ -0,0 +1,438 @@
+		}
+	}
+	@-moz-keyframes md-bounce-2 {
+		0% {
+			opacity: 0;
+			-moz-transform: translateX(-100px);
+		}
+		60% {
+			opacity: 1;
+			-moz-transform: translateX(20px);
+		}
+		100% {
+			-moz-transform: translateX(0);
+		}
+	}
+	@keyframes md-bounce-2 {
+		0% {
+			opacity: 0;
+			-webkit-transform: translateX(-100px);
+			-moz-transform: translateX(-100px);
+			-ms-transform: translateX(-100px);
+			-o-transform: translateX(-100px);
+			transform: translateX(-100px);
+		}
+		60% {
+			opacity: 1;
+			-webkit-transform: translateX(20px);
+			-moz-transform: translateX(20px);
+			-ms-transform: translateX(20px);
+			-o-transform: translateX(20px);
+			transform: translateX(20px);
+		}
+		100% {
+			-webkit-transform: translateX(0);
+			-moz-transform: translateX(0);
+			-ms-transform: translateX(0);
+			-o-transform: translateX(0);
+			transform: translateX(0);
+		}
+	}
+	@-webkit-keyframes md-bounce-2-inverse {
+		0% {
+			opacity: 0;
+			-webkit-transform: translateX(100px);
+		}
+		60% {
+			opacity: 1;
+			-webkit-transform: translateX(-20px);
+		}
+		100% {
+			-webkit-transform: translateX(0);
+		}
+	}
+	@-moz-keyframes md-bounce-2-inverse {
+		0% {
+			opacity: 0;
+			-moz-transform: translateX(100px);
+		}
+		60% {
+			opacity: 1;
+			-moz-transform: translateX(-20px);
+		}
+		100% {
+			-moz-transform: translateX(0);
+		}
+	}
+	@keyframes md-bounce-2-inverse {
+		0% {
+			opacity: 0;
+			-webkit-transform: translateX(100px);
+			-moz-transform: translateX(100px);
+			-ms-transform: translateX(100px);
+			-o-transform: translateX(100px);
+			transform: translateX(100px);
+		}
+		60% {
+			opacity: 1;
+			-webkit-transform: translateX(-20px);
+			-moz-transform: translateX(-20px);
+			-ms-transform: translateX(-20px);
+			-o-transform: translateX(-20px);
+			transform: translateX(-20px);
+		}
+		100% {
+			-webkit-transform: translateX(0);
+			-moz-transform: translateX(0);
+			-ms-transform: translateX(0);
+			-o-transform: translateX(0);
+			transform: translateX(0);
+		}
+	}
+</style>
+
+<section id="md-timeline" class="md-container">
+   	<% if (site.posts.length) { -%>
+   		<% site.posts.sort('date', -1).limit(site.posts.length).each(function(post){ %>
+			<div class="md-timeline-block">
+                		<div class="md-timeline-date blue">
+                   			<div class="md-date"><%= date(post.date, 'MMM DD YYYY') %></div>
+                		</div>
+
+                		<div class="md-timeline-content">
+                    			<div class="md-timeline-title"><a href="<%- url_for(post.path) %>"><%= post.title %></a></div>
+					<div class="md-timeline-info">
+						<span class="md-timeline-info-span">Tags:
+							<%- list_tags(post.tags, {
+								show_count: false,
+								style: 'none'
+							}) %>
+						</span>
+						<span class="md-timeline-info-span">Categorias:
+							<%- list_categories(post.categories, {
+								show_count: false, 
+								style: 'none'
+							}) %>
+						</span>
+					</div>
```

Comment:
Sorry for that. I just copied the CSS from the [typecho repo](https://github.com/viosey/typecho-theme-material). I think there are still some issues in the style.

I tried to fix the float layout problem. I did several attempts but were not satisfied. [This commit](https://github.com/viosey/hexo-theme-material/pull/253/commits/548fea514b24e45643ed82e9a0458edad66d0fd6) may be a acceptable modification. I hope someone who knows more CSS styles than me would prettify it.

Finally, I recommend that use fewer words to describe the tags and categories, which means you won't meet the above issue.